### DISCUSSION TITLE
Issue regarding Vanilla JavaScript template.

### DISCUSSION THREAD

**Discussion by Schnarfengargle123** [codesandbox/codesandbox-client#6616] [Q&A] [answered]
Hi folks,

I'm having some issues with CodeSandbox where the `this` keyword, is returning `undefined`, when it ought not to be. I'm wondering what exactly is causing this issue. My attatched screenshot demonstrates the problem.
<img width="1680" alt="Screenshot 2022-04-10 at 15 28 11" src="https://user-images.githubusercontent.com/11248167/162624558-5e1c4fd4-96e5-4df9-958b-a7faee16be2c.png">

Thanks!

**Comment by JamesACS** [comment, marked as answer]
Hey @Schnarfengargle123 - This seems to be occurring when you call the whatsThis function, it's likely to do with CodeSandbox having JS in strict mode by default, which means that undeclared variables aren't automatically global. 

You're also using var, which is an outdated keyword - let and const are more modern! 

If you move the var a within the function, it begins working
